src/app/modules/main/chat_section/controller.nim

```diff
@@ -647,6 +651,25 @@ proc ownsCollectible*(self: Controller, chainId: int, contractAddress: string, t
 
   return false
 
+proc ownsENS*(self: Controller, ensPattern: string): bool =
+  let addresses = self.walletAccountService.getWalletAccounts().filter(a => a.walletType != WalletTypeWatch).map(a => a.address)
+  let ensNames = self.ensService.reverseResolveENS(addresses)
```

Comment:
I wonder if we should simply use `getAllMyEnsUsernames` here.
We already have the list of connected ENS usernames in the database, it seems to make sense to use it.

Though there will be a behaviour difference. 
User might have an ENS associated with given wallet address, which could have been not connected to the app in `Settings -> ENS usernames`. 

I don't know which behaviour is required, but I'd vote for using only connected usernames. At the same time we could add corresponding UI feature in `Settings -> ENS Usernames` to search for owned ENS usernames and add them to the database.

Reply:
Started a discussion around it here https://github.com/status-im/status-go/pull/3467#issuecomment-1540027452